matrix_client/api.py

```diff
@@ -294,6 +294,94 @@ def set_room_topic(self, room_id, topic):
         }
         return self._send("PUT", "/rooms/" + room_id + "/state/m.room.topic", body)
 
+    def get_power_levels(self, room_id):
+        """Perform GET /rooms/$room_id/state/m.room.power_levels
+        Args:
+            room_id(str): The room ID
+        """
+        return self._send("GET", "/rooms/" + room_id + "/state/m.room.power_levels")
+
+    def modify_user_power_levels(self, room_id, users=None, users_default=None):
+        """Perform GET and PUT /rooms/$room_id/state/m.room.power_levels to
+            modify users' power levels
+        Args:
+            room_id(str): The room ID
+            users(dict): Power levels to assign to specific users, in the form
+                {"@name0:host0": 10, "@name1:host1": 100, "@name3:host3", None}
+                A level of None causes the user to revert to the default level
+                as specified by users_default.
+            users_default(int): Default power level for users in the room
+        """
+        body = self.get_power_levels(room_id)
+        if users_default:
+            body["users_default"] = users_default
+
+        if users:
+            if "users" in body:
+                body["users"].update(users)
+            else:
+                body["users"] = users
+
+            # Remove any keys with value None
+            for user in list(body["users"].keys()):
+                if body["users"][user] is None:
+                    del body["users"][user]
+        return self._send("PUT", "/rooms/" + room_id +
+                          "/state/m.room.power_levels", body)
+
+    def modify_required_power_levels(self, room_id, events=None,
```

Comment:
I'd prefer it if the arguments for the levels were ``**kwargs``. Generally speaking if it's more than 3 or 4 args then you really need to include that over lots of ``foo=None, bar=None, ..``. As you can see, it gets messy very quickly.